pkg/status/types.go

```diff
@@ -5,13 +5,35 @@
 
 package status
 
+import (
+	"encoding/json"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+)
+
 // CLCChecks is used to unmarshall the runner expvar payload for CLC Runner
 type CLCChecks struct {
 	Checks map[string]map[string]CLCStats `json:"Checks"`
 }
 
 // CLCStats is used to unmarshall the stats needed from the runner expvar payload
 type CLCStats struct {
-	AverageExecutionTime int `json:"AverageExecutionTime"`
-	MetricSamples        int `json:"MetricSamples"`
+	AverageExecutionTime int  `json:"AverageExecutionTime"`
+	MetricSamples        int  `json:"MetricSamples"`
+	LastExecFailed       bool `json:"LastExecFailed"`
+}
+
+// UnmarshalJSON overwrites the unmarshall method for CLCStats
+func (d *CLCStats) UnmarshalJSON(data []byte) error {
+	var stats check.Stats
+	if err := json.Unmarshal(data, &stats); err != nil {
+		return err
+	}
+	d.AverageExecutionTime = int(stats.AverageExecutionTime)
+	d.MetricSamples = int(stats.MetricSamples)
+	if stats.LastError != "" {
+		d.LastExecFailed = true
```

Comment:
If you call this method twice, and there is no error,  `d.LastExecFailed` keeps its older value which can be `true`.

Reply:
in theory this shouldn't  happen with the current usage of this function. But yes I agree in can happen in the futur